File: rust-resend-examples/examples/domains.rs
use resend_rs::Resend;

#[tokio::main]
async fn main() {
    dotenvy::dotenv().ok();

    let api_key = std::env::var("RESEND_API_KEY").expect("RESEND_API_KEY environment variable is required");

    let resend = Resend::new(&api_key);

    // 1. List all domains
    println!("=== Listing Domains ===");
    let domains = match resend.domains.list().await {
        Ok(d) => d,
        Err(e) => {
            eprintln!("Error listing domains: {}", e);
            std::process::exit(1);
        }
    };

    println!("Found {} domain(s)", domains.data.len());
    for domain in &domains.data {
        println!("  - {} (status: {}, id: {})", domain.name, domain.status, domain.id);
    }

    // 2. Get domain details (if any exist)
    if let Some(first) = domains.data.first() {
        let domain_id = &first.id;

        println!("\n=== Domain Details: {} ===", first.name);
        match resend.domains.get(domain_id).await {
            Ok(domain) => {
                println!("Name: {}", domain.name);
                println!("Status: {}", domain.status);
                println!("Region: {}", domain.region);
                println!("Created: {}", domain.created_at);

                if !domain.records.is_empty() {
                    println!("\nDNS Records:");
                    for record in &domain.records {
                        println!("  {}: {} -> {}", record.record_type, record.name, record.value);
                    }
                }
            }
            Err(e) => {
                eprintln!("Error getting domain: {}", e);
            }
        }

        // 3. Verify domain
        println!("\n=== Verifying Domain ===");
        match resend.domains.verify(domain_id).await {
            Ok(_) => println!("Domain verification initiated!"),
            Err(e) => println!("Verification request sent: {}", e),
        }
    }

    // To create a new domain (uncomment):
    // let params = CreateDomainOptions::new("mail.example.com");
    // let new_domain = resend.domains.create(params).await;

    // To delete a domain (uncomment):
    // resend.domains.remove(domain_id).await;

    println!("\nDone!");
}

File: bun-resend-examples/javascript/examples/domains.js
import { Resend } from "resend";

const resend = new Resend(process.env.RESEND_API_KEY);

// 1. List all domains
console.log("=== Listing Domains ===");
const { data: domains } = await resend.domains.list();
console.log(`Found ${domains?.data.length || 0} domain(s)`);

domains?.data.forEach((domain) => {
  console.log(`  - ${domain.name} (status: ${domain.status}, id: ${domain.id})`);
});

// 2. Get domain details (if any exist)
if (domains?.data.length) {
  const domainId = domains.data[0].id;

  console.log(`\n=== Domain Details: ${domains.data[0].name} ===`);
  const { data: domain } = await resend.domains.get(domainId);

  if (domain) {
    console.log("Name:", domain.name);
    console.log("Status:", domain.status);
    console.log("Region:", domain.region);
    console.log("Created:", domain.created_at);

    if (domain.records?.length) {
      console.log("\nDNS Records:");
      domain.records.forEach((record) => {
        console.log(`  ${record.type}: ${record.name} -> ${record.value}`);
      });
    }
  }

  // 3. Verify domain
  console.log("\n=== Verifying Domain ===");
  await resend.domains.verify(domainId);
  console.log("Domain verification initiated!");
}

console.log("\nDone!");
